### PR TITLE
Nikos/bunch of v5 changes

### PR DESCRIPTION
**Pull request by nikosdouvlis**
## Description
- Rename beta-v5 to beta
- Port the cookie-setting strategy change from v4
- Stop adding "redirect" as authreason as it overrides AuthReason in some scenarios, making debugging harder 

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: .changeset/pre.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "mode": "pre",
-  "tag": "beta-v5",
+  "tag": "beta",
```
Comment (dimkl) [inline .changeset/pre.json:3]: 🤔 Currently all the package.json files in this monorepo refer to `@beta-v5.{something}`. Should we add a patch version bump in this changeset to make them convert to `beta`?

File: packages/clerk-js/src/core/resources/Session.ts [hunk 2]
```diff
@@ -160,7 +160,11 @@ export class Session extends BaseResource implements SessionResource {
     const cachedEntry = skipCache ? undefined : SessionTokenCache.get({ tokenId }, leewayInSeconds);
 
     if (cachedEntry) {
-      return cachedEntry.tokenResolver.then(res => res.getRawString());
+      const cachedToken = await cachedEntry.tokenResolver.then(res => res);
+      if (!template) {
+        eventBus.dispatch(events.TokenUpdate, { token: cachedToken });
+      }
+      return cachedToken.getRawString();
```
Comment (dimkl) [inline packages/clerk-js/src/core/resources/Session.ts:167]: 🤔 This means we forgot to apply Bryce change to the beta version?
ref: https://github.com/clerk/javascript/pull/2585

**Review by dimkl — APPROVED** [review]

File: packages/clerk-js/src/core/resources/Session.ts (hunk 2 above)
Comment (nikosdouvlis) [reply packages/clerk-js/src/core/resources/Session.ts:167]: We did not "forget" per se, as it was not necessarily needed but I decided to port it for two reasons:
- the vendor-level protection will be around for ~a year so there's a small chance that someone upgrading to v5 might break some of their users after deploying to a vendor that still has the protections active
- its going to be easier for us to debug any related issues that come up if both versions share the same implementation while people are upgrading

File: .changeset/pre.json (hunk 1 above)
Comment (nikosdouvlis) [reply .changeset/pre.json:3]: I think this is going to happen in the next Vesion Packages PR even if we don't do anything :)

**Pull request merged**